ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Flex, FlexItem, PageSection } from '@patternfly/react-core';
+
+import useURLSearch from 'hooks/useURLSearch';
+import { fetchClustersAsArray } from 'services/ClustersService';
+import { Cluster } from 'types/cluster.proto';
+import ClusterSelect from './ClusterSelect';
+import NamespaceSelect from './NamespaceSelect';
+
+export type Namespace = {
+    metadata: {
+        id: string;
+        name: string;
+    };
+};
+
+function NetworkGraphPage() {
+    const [clusters, setClusters] = React.useState<Cluster[]>([]);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const selectedClusterId = (searchFilter.Cluster as string) || '';
+    const selectedNamespaces = (searchFilter.Namespace as string[]) || [];
+
+    React.useEffect(() => {
+        fetchClustersAsArray()
+            .then((data) => {
+                setClusters(data as Cluster[]);
+            })
+            .catch(() => {
+                // TODO
+            });
+    }, []);
+
+    function updateSelectedClusterId(selection) {
```

Comment:
Since we only have one cluster, I was wondering what would happen if we did the following:
1. Selected cluster `A`
2. Selected namespace `1`, `2`, `3`
3. Selected a different cluster `B`

I think I would expect the `selectedNamespaces` to be reset too. From the code it looks like the Namespace Select will fetch the namespaces for the new cluster, but I think the selection would still be for the previous namespaces

Reply:
Looking ahead, I think we'd have to make sure we only show the graph if both the cluster and namespaces are selected. If only the cluster is selected or we pick a new cluster (clearing the namespaces as a result), we wouldn't show the graph + side panel. I think we already do that in the current network graph though